README
Dist::Zilla::Util::BundleInfo



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Util-BundleInfo

1.001006 2017-03-04T04:27:29Z 2fd0932

1.001005 2017-03-04T03:56:20Z 4a27294
 [BugFix / 2]
 - Ensure $INC{} is populated on old moose installs
 - Closes https://github.com/kentnl/Dist-Zilla-Util-BundleInfo/issues/2

 [BugFix / 3]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Util-BundleInfo/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 1.001004, see misc/*.deps* for details
 - configure: (suggests: +1)
 - develop: +2 ↑1 -1 (suggests: ↑1)
 - test: ↓1

1.001004 2015-12-20T12:04:26Z e899016
 [Bugfix]
 - Arrays of 1 now work the same way scalars do for serialization to ini.
 - Arrays of 0 now dont emit anything for serialization to ini ( and warns )
 - Non-Multivalue args with aliases now work.

 [Dependencies::Stats]
 - Dependencies changed since 1.001003, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +15 ↑2 -4 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1
 - test: +6 (recommends: +1 ↑1 ↓1)

1.001003 2014-08-15T21:50:53Z
 [00 Trivial]
 - CPANDAY!
 - No Code Changes.
 - Tests augmented.
 - Metadata updated.
 - Whitespace adjustment.

 [Dependencies::Stats]
 - Dependencies changed since 1.001002, see misc/*.deps* for details
 - configure: ↓1
 - develop: +1 (suggests: ↑1)
 - test: -4 (recommends: ↑1 -2)

 [Metadata]
 - add 1x contributor.

 [Misc]
 - Updated POD in source side to produce less whitespace at top of generated code.

 [Tests]
 - ReportVersion test update.

1.001002 2014-07-23T19:11:00Z
 [00 Trivial]
 - Misc packaging cleanups

 [Dependencies::Stats]
 - Dependencies changed since 1.001001, see misc/*.deps* for details
 - develop: +1 -1 (suggests: +2 -1)
 - test: +2 ↑1 (recommends: +1)

1.001001 2014-06-02T22:21:02Z
 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - develop: ↑1

 [Documentation]
 - POD Indentation/Style tweaked for readability.

1.001000 2014-06-02T09:06:39Z
 [01 Plugin::payload_list]
 - New function returns payload contents in list form with MVP checked arity.

1.000000 2014-06-02T02:30:43Z
 [01 MAJOR]
 - Mostly packaging changes that will predominantly be significant for non-CPAN toolchains.

 [02 Migrated to EUMM]
 - ExtUtils::MakeMaker is now the installer of choice.
 - "Latest" dependencies moved to only recommends instead of requires.

 [03 New Version Scheme]
 - Moving from v-versions (x.y.z) to reals (x.yyyzzz)
 - This is the primary reason for the 1.x on the box.

 [Dependencies::Stats]
 - Dependencies changed since 0.1.3, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +42 ↑1 -2 (recommends: -1, suggests: +1 -1)
 - runtime: +1 ↓1
 - test: +5 ↓1 -3 (recommends: +4)

0.1.3 2013-09-25T18:02:50Z
 [Bugfix]
 - Update Test::Compile to work on Win32

 [Dependencies::Stats]
 - Dependencies changed since 0.1.2, see misc/*.deps* for details
 - test: +1

0.1.2 2013-08-31T03:20:53Z
 [!Minor]
 - Excise old Test::Compile

 [Dependencies::Stats]
 - Dependencies changed since 0.1.1, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - test: +2 -2

0.1.1 2013-08-01T17:20:02Z
 [Dependencies::Stats]
 - Dependencies changed since 0.1.0, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - test: +2 -2

 [Documentation]
 - Add MetaPOD annotations

0.1.0 2013-08-01T12:20:06Z
 - First version.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4007"
  },
  "configure_requires" => {
    "Module::Build" => "0.4007"
  },
  "dist_abstract" => "Load and interpret a bundle",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Util-BundleInfo",
  "dist_version" => "0.1.4",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Util::BundleInfo",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Dist::Zilla::Util" => 0,
    "Module::Runtime" => 0,
    "Moo" => "1.000008",
    "perl" => "v5.10.0",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla::PluginBundle::Basic" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.98"
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Load and interpret a bundle",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Util-BundleInfo",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Util::BundleInfo",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla::Util" => 0,
    "Module::Runtime" => 0,
    "Moo" => "1.000008",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::PluginBundle::Basic" => 0,
    "Dist::Zilla::PluginBundle::Classic" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::PluginBundle" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Moose" => 0,
    "Test::More" => 0,
    "Test::Warnings" => 0
  },
  "VERSION" => "1.001006",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::Plugin::GatherDir" => 0,
  "Dist::Zilla::PluginBundle::Basic" => 0,
  "Dist::Zilla::PluginBundle::Classic" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Role::PluginBundle" => 0,
  "Dist::Zilla::Util" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Runtime" => 0,
  "Moo" => "1.000008",
  "Moose" => 0,
  "Test::More" => 0,
  "Test::Warnings" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
